metomi/rose/suite_run.py

```diff
@@ -233,46 +250,52 @@ def run_impl(self, opts, args, uuid, work_files):
                 os.chdir(suite_dir)
 
             # Housekeep log files
+            ## remove:obsolete (not re-implementing this in Cylc)
             now_str = None
             if not opts.install_only_mode and not opts.local_install_only_mode:
                 now_str = datetime.utcnow().strftime("%Y%m%dT%H%M%SZ")
                 self._run_init_dir_log(opts, now_str)
             self.fs_util.makedirs("log/suite")
 
             # Rose configuration and version logs
+            ## keep
             self.fs_util.makedirs("log/rose-conf")
             run_mode = opts.run_mode
             if run_mode not in ["reload", "restart", "run"]:
                 run_mode = "run"
             mode = run_mode
-            if opts.validate_suite_only_mode:
-                mode = "validate-suite-only"
-            elif opts.install_only_mode:
-                mode = "install-only"
-            elif opts.local_install_only_mode:
-                mode = "local-install-only"
+            #if opts.validate_suite_only_mode:
+            #    mode = "validate-suite-only"
+            #elif opts.install_only_mode:
+            #    mode = "install-only"
+            #elif opts.local_install_only_mode:
+            #    mode = "local-install-only"
             prefix = "rose-conf/%s-%s" % (strftime("%Y%m%dT%H%M%S"), mode)
 
             # Dump the actual configuration as rose-suite-run.conf
             ConfigDumper()(conf_tree.node, "log/" + prefix + ".conf")
 
             # Install version information file
+            ## keep
             write_source_vc_info(
                 suite_conf_dir, "log/" + prefix + ".version", self.popen)
 
             # If run through rose-stem, install version information
             # files for each source tree if they're a working copy
+            ## keep
```

Comment:
@dpmatthews has suggested we remove this from Rose and build it directly into Cylc.

This could be an additional Cylc plugin with git/svn support.